### PR TITLE
Add CI backend

### PR DESCRIPTION
**Pull request by Keyshal-211**
,## Description of changes
Add CI backend including building the backend, running the tests and building the docker container whenever something is pushed to the repository.

  
## How has this been tested?

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test

Imho, no testing is needed for this issue as we can see that the pipeline is running and showing green lights for the template micro service. 
    
## Checklist before requesting a review
- [x] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas (there are no such areas)
- [x] My code fulfills all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [x] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build the docker image
+        run: docker-compose build
+
+      - name: up postgresql container runs
+        run: docker-compose up -d
```
Comment (PaulBredl) [inline .github/workflows/test.yaml:14]: this will build and run the application already in the docker container. Ideally we only start the "testDb"  container, which I'm sure is possible with a command.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Build with Gradle
+        uses: gradle/gradle-build-action@749f47bda3e44aa060e82d7b3ef7e40d953bd629
+        with:
+          arguments: build
+      - name: Execute tests
+        uses: gradle/gradle-build-action@749f47bda3e44aa060e82d7b3ef7e40d953bd629
+        with:
+          arguments: test
+
+  docker-build:
```
Comment (PaulBredl) [inline .github/workflows/test.yaml:33]: What exactly does this do? Does it start the app in the docker container?

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Run chmod to make gradlew executable
+        run:  chmod +x gradlew
+      - name: Build with Gradle
+        uses: gradle/gradle-build-action@749f47bda3e44aa060e82d7b3ef7e40d953bd629
+        with:
+          arguments: build
+      - name: Execute tests
+        uses: gradle/gradle-build-action@749f47bda3e44aa060e82d7b3ef7e40d953bd629
+        with:
+          arguments: test
```
Comment (PaulBredl) [inline .github/workflows/test.yaml:31]: It would be nice to get like results from this step, like a test report or something, but that's not part of this issue imo but something for a later sprint. (I don't know if that is even possible)

**Review by PaulBredl — CHANGES_REQUESTED** [review]
I think we should not stat the whole docker container in the build step, as it is kinda redundant, but only the testDb container.

File: .github/workflows/test.yaml (hunk 1 above)
Comment (PaulBredl) [reply .github/workflows/test.yaml:14]: It should be "docker compose up -d testDb"

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Run chmod to make gradlew executable
+        run:  chmod +x gradlew
+      - name: Build with Gradle
+        uses: gradle/gradle-build-action@749f47bda3e44aa060e82d7b3ef7e40d953bd629
+        with:
+          arguments: build
+      - name: Execute tests
+        uses: gradle/gradle-build-action@749f47bda3e44aa060e82d7b3ef7e40d953bd629
+        with:
+          arguments: cleanTest test
```
Comment (PaulBredl) [inline .github/workflows/test.yaml:31]: I added "cleanTest" to force gradle to always execute even if no files changed.

File: .github/workflows/test.yaml (hunk 1 above)
Comment (bergdola) [reply .github/workflows/test.yaml:14]: Unfortunately, that didn't work..

Comment (PaulBredl) [reply .github/workflows/test.yaml:14]: Then please note that as a to-do for later
